### PR TITLE
Jt/seperate player spawns

### PR DESCRIPTION
**Pull request by jnterry**
This ensures different members of the party spawn on different spawn pads  when starting the level

### CODE REVIEW

**Comment by grant-m-s** [comment]
So trying to run 3 clients, first 2 work correctly 3rd one doesn't start it gets stuck after ```Finished processing Entity with server ID 1514 and 9 components and validity 1``` then nothing renders. Could be just my machine though, anyone else?

**Comment by AnthonyWharton** [comment]
Yeah, Afraid this seems to hang for me with larger number of clients. Sorry old bean :/

File: src/lib-tempo/src/entity/EntityCreationServer.cpp [hunk 1]
```diff
@@ -27,8 +27,25 @@ anax::Entity newPlayer(anax::World &world, uint32_t party_number)
...
-	entity_player.addComponent<tempo::ComponentStagePosition>(glm::ivec2(10 + (party_number * fheight), 0));
+	auto& comp_party = entity_player.addComponent<tempo::ComponentParty>(party_number);
+
+	glm::ivec2 player_spawn_location = glm::ivec2(10 + (party_number * fheight), 0);
+
+	// Modify the spawn location based on the entity index within the party
+	// This is hard coded but relies on the level design!
+	switch(comp_party.entity_index){
+
+	case 1: player_spawn_location += glm::ivec2(- 2,  0); break;
```
Comment (qqii) [inline src/lib-tempo/src/entity/EntityCreationServer.cpp]: What are this going to be for multiple feeder areas.

**Comment by jnterry** [comment]
This is now a big hack - players are allocated slots on round robin basis rather than having server keep track of which slots are occupied.

This works fine if 6 clients join in sequence, if we get crashed clients and then people rejoining it may be that slots are reused, but players should still be fairly spread out.

Either way it seems to work for arbitary number of clients. I'll see if I can get proper slot allocation to work, but this will probably do for tomorrow if nothing else works.

**Pull request merged**